mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
 package org.openhab.habdroid.model;
 
+import net.bytebuddy.implementation.bytecode.Throw;
+
 import org.json.JSONObject;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.runners.MockitoJUnitRunner;
 
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutionException;
+
 import static junit.framework.Assert.assertEquals;
 import static junit.framework.Assert.assertTrue;
 import static org.mockito.Mockito.mock;
 
 @RunWith(MockitoJUnitRunner.class)
 public class OpenHAB2WidgetTest {
-    OpenHABWidget rootWidget = mock(OpenHAB2Widget.class);
+    private OpenHABWidget rootWidget = mock(OpenHAB2Widget.class);
+    OpenHABWidget sut1;
+    OpenHABWidget sut2;
+    OpenHABWidget sut3;
 
     @Test
-    public void getIconPath_iconExists_returnIconUrlToIconServlet() throws Exception {
-        OpenHABWidget sut = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(), "PNG");
-        assertEquals("icon/groupicon?state=OFF&format=PNG", sut.getIconPath());
+    @Before
+    public void testCreateOpenHABWidgetFromJson_createsOpenHAB2Widget() throws Exception {
+        sut1 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(1), "PNG");
+        sut2 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(2), "SVG");
+        sut3 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(3), "SVG");
+        assertTrue(sut1 instanceof OpenHAB2Widget);
```

Comment:
I personally dislike asserts in before(class)/after(class), as this is setup code, that will possibly run multiple times. I would create separate tests for them :)

Reply:
Moved them to a new test